src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/SimpleResourceHandler.java

```diff
@@ -0,0 +1,261 @@
+     */
+    public int insert(T resource, int amount, TransferAction actionType) {
+        try (Transaction transaction = TransactionManager.open(null)) {
+            int inserted = handler.insert(resource, amount, transaction);
+            actionType.commit(transaction);
+            return inserted;
+        }
+    }
+
+    /**
+     * Extracts a given amount of the resource from the handler at the given index.
+     *
+     * @param index      The index to extract the resource from. <strong>Must be Non-Negative</strong>
+     * @param resource   The resource to extract.
+     * @param amount     The amount of the resource to extract. <strong>Must be Non-Negative</strong>
+     * @param actionType The action to take, whether to simulate the insertion or actually perform it.
+     * @return The amount (<strong>Must be Non-Negative</strong>) of the resource that was (or would have been, if simulated) extracted.
+     */
+    public int extract(int index, T resource, int amount, TransferAction actionType) {
+        try (Transaction transaction = TransactionManager.open(null)) {
+
+            int extracted = handler.extract(index, resource, amount, transaction);
+            actionType.commit(transaction);
+            return extracted;
+        }
+    }
+
+    /**
+     * Extracts a given amount of the resource from the handler. Distribution of the resource is up to the handler.
+     * <p>
+     * Implementation advice, don't just have this call {@link #extract(int, IResource, int, TransferAction)}, as you may needlessly re-check validations.
+     * See {@link ResourceStorageComponentHandler#extract(IResource, int, TransactionContext) ResourceStorage.extractBehaviour} for an example.
+     *
+     * @param resource   The resource to extract.
+     * @param amount     The amount of the resource to extract. <strong>Must be Non-Negative</strong>
+     * @param actionType The action to take, whether to simulate the insertion or actually perform it.
+     * @return The amount (<strong>Must be Non-Negative</strong>) of the resource that was (or would have been, if simulated) extracted.
+     */
+    public int extract(T resource, int amount, TransferAction actionType) {
+        try (Transaction transaction = TransactionManager.open(null)) {
+            int extracted = handler.extract(resource, amount, transaction);
+            actionType.commit(transaction);
+            return extracted;
+        }
+    }
+
+    public IResourceHandler<T> handler() {
+        return handler;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) return true;
+        if (obj == null || obj.getClass() != this.getClass()) return false;
+        var that = (SimpleResourceHandler) obj;
+        return Objects.equals(this.handler, that.handler);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(handler);
```

Comment:
Just return `handler.hashCode()`, no need to do vararg array creation for a single element